### PR TITLE
Add a dune file and a new package extlib_wrapped 

### PR DESCRIPTION
**Pull request by jorisgio**
This intends to solve same issue as #53

### CODE REVIEW

**Comment by olafhering** [comment]
I think the changes from dune-universe/ocaml-extlib@4e178dc0616add7836909d474ef86f8757d13182 to build with `dune` are more complete. Still, they lack support for the `dune runtest` target.

@ygrek what is your opinion regarding building the entire thing with dune?

**Comment by avery-laird** [comment]
This still seems to be a problem with `dune utop`:

```
[<me>@<me> js2sil]$ dune utop
File "<home>/.opam/ocaml-variants.4.07.1+flambda/lib/base64/base64.cma(Base64)", line 1:
Warning 31: files <home>/.opam/ocaml-variants.4.07.1+flambda/lib/base64/base64.cma(Base64) and <home>/.opam/ocaml-variants.4.07.1+flambda/lib/extlib/extLib.cma(Base64) both define a module named Base64
```

**Comment by ygrek** [comment]
> @ygrek what is your opinion regarding building the entire thing with dune?

I am ok to add it in parallel to Makefile.

**Comment by Et7f3** [comment]
What is the state of this pr ? Can I help ?

File: .merlin [hunk 1]
```diff
@@ -1,4 +1,22 @@
+EXCLUDE_QUERY_DIR
```
Comment (Et7f3) [inline .merlin:1]: should be ignored by git

File: dune [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ (name genconfig)
+ (modules genconfig)
+ (libraries dune.configurator))
+
+(rule
+ (targets cppo_flags)
+ (action
+  (run ./genconfig.exe)))
+
+(copy_files src/*.ml*)
```
Comment (Et7f3) [inline dune:19]: you can declare this dune stanza in src folder and avoid copy.

File: dune [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+(library
+ (name extlib)
+ (public_name extlib_wrapped)
```
Comment (Et7f3) [inline dune:23]: why this breakage ? if named extLib then user could still consume without noticing the switch to dune.
```suggestion
 (name extLib)
 (public_name extlib)
```

File: extlib_wrapped.opam [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "Bardur Arantsson"
+  "Janne Hellsten"
+  "Richard W.M. Jones"
+  "ygrek"
+  "Gabriel Scherer"
+  "Pietro Abate"
+]
+build: ["dune" "build" "-p" name]
+depends: [
+  "ocaml"
```
Comment (Et7f3) [inline extlib_wrapped.opam:25]: dune is obviously omitted :eyes: their is `dune_generate_opam_file` if you want.

File: flags.sexp [hunk 5]
```diff
@@ -0,0 +1 @@
+(-w -3-6-9-27-32-33-39-50 -no-strict-sequence)
```
Comment (Et7f3) [inline flags.sexp:1]: shouldn't be committed

**Comment by ygrek** [comment]
Superceded by #59, thanks! After switching to dune will make a breaking change with version bump  to the wrapped library.

**Pull request closed without merging**